src/rin2016/module8/associative_array/AssociativeArray.java

```diff
@@ -0,0 +1,71 @@
+package rin2016.module8.associative_array;
+
+public class AssociativeArray {
+	ElementArray[] _groups;
+	private int _size;
+	private static final int DEFAULT_SIZE = 10;
+	private static final int SEARCH_ERROR = -1;
+	
+	public AssociativeArray(int counterGroups) {
+		if(counterGroups <= 0){
+			counterGroups = DEFAULT_SIZE;
+		} 
+		_groups = new ElementArray[counterGroups];
+	}
+	
+	public void add(final ElementArray group) {
+		if(_size >= _groups.length) {
+			_groups = increaseSize();
+		}
+		_groups[_size] = group.getCopyElementArray();
+		_size++;
+	}
+	
+	private ElementArray[] increaseSize() {
+		ElementArray[] temp = new ElementArray[_groups.length+10];
+		for (int i = 0; i < _size; i++) {
+			temp[i] = _groups[i];
+		}
+		return temp;
+	}

```

Comment:
This method name is not clear. Why does increaseSize() return ElementArray[]? Could it be void? If it would be void then its name is good.